relay-server/src/actors/outcome.rs

```diff
@@ -142,9 +143,11 @@ pub enum OutcomeType {
 /// Defines the possible outcomes from processing an event.
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
 pub enum Outcome {
-    /// The outcome state "Accepted" is never emitted by Relay as the event may be discarded
-    /// by the processing pipeline after Relay.
-    /// Only the `save_event` task in Sentry finally accepts an event.
+    /// The event has been accepted and handled completely.
+    ///
+    /// This is never emitted by Relay as the event may be discarded by the processing pipeline
+    /// after Relay. Only the `save_event` task in Sentry finally accepts an event.
+    Accepted,
```

Comment:
This brings back the `Accepted` outcome variant. The enum here in Relay is the canonical place for documentation and should always be complete. This is the same as with `Abuse` below. Note that the doc comment was also associated to the wrong variant as a result.

I hope that I brought back all required logic for handling `Accepted` correctly. However, we should now prevent emitting it:
 1. Guard against accidentally logging an accepted outcome in our own code. I would propose to do that in the `TrackOutcome` (not raw!) message handler and drop the outcome along with logging an error message to Sentry. Then, continue operation without failing.
2. Reject `Accepted` outcomes from downstream clients. I'm not sure where the best place for this is, probably the handler of the outcomes endpoint.

Reply:
Accepted was removed here: https://github.com/getsentry/relay/pull/1119/files#diff-53567d8af9f1bfd9cbef8fd542922b8cc93dee466d6ab327abb3e704c2649ac1L102

Reply:
Those two should cover it, yes. Client reports cannot contain accepted outcomes since their outcome type is encoded into the object key, like `discarded_events` or `rate_limited_events`.

Reply:
Thanks. As discussed offline, I'll move the changes around `Outcome::Accepted` into a dedicated PR.